net-vpn/networkmanager-openconnect/networkmanager-openconnect-1.2.10.ebuild

```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+GNOME_ORG_MODULE="NetworkManager-${PN##*-}"
+
+inherit gnome2
+
+DESCRIPTION="NetworkManager OpenConnect plugin"
+HOMEPAGE="https://wiki.gnome.org/Projects/NetworkManager https://gitlab.gnome.org/GNOME/NetworkManager-openconnect"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc64 ~riscv ~x86"
+IUSE="gtk"
+
+DEPEND="
+	>=net-misc/networkmanager-1.2:=
+	>=dev-libs/glib-2.34:2
+	>=dev-libs/dbus-glib-0.74
+	dev-libs/libxml2:2
+	dev-libs/glib:2
+	>=net-vpn/openconnect-3.02:=
+	gtk? (
+		>=app-crypt/libsecret-0.18
+
+		>=app-crypt/gcr-3.4:0=
+		>=x11-libs/gtk+-3.12:3
+
+		>=gui-libs/gtk-4.0:4
+		>=net-libs/libnma-1.8.36
+		net-libs/webkit-gtk:4.1
```

Comment:
````
if test x"$with_gtk4" != xno; then
	PKG_CHECK_MODULES(GTK4, gtk4 >= 4.0)
	GTK4_CFLAGS="$GTK4_CFLAGS -DGDK_VERSION_MIN_REQUIRED=GDK_VERSION_4_0"
	GTK4_CFLAGS="$GTK4_CFLAGS -DGDK_VERSION_MAX_ALLOWED=GDK_VERSION_4_0"
	PKG_CHECK_MODULES(LIBNMA_GTK4, libnma-gtk4 >= 1.8.33)
fi

PKG_CHECK_MODULES(WEBKIT, webkit2gtk-4.1, [wekbit=4.1],
			  [PKG_CHECK_MODULES(WEBKIT, webkit2gtk-4.0,
			  [webkit=4.0], AC_MSG_ERROR(Neither webkit2gtk-4.0 nor wekit2gtk-4.1 found))])

````

It looks to me like webkit-gtk is required unconditionally. I assume that's a bug in `configure.ac` and it should only be required GTK is enabled. Will have to look at the git history.

Reply:
Looks like it was added as an unconditional dep in https://gitlab.gnome.org/GNOME/NetworkManager-openconnect/-/commit/1e86b38

Reply:
So it definitely looks like `configure.ac`'s check for webkit should be moved inside an authdlg conditional, probably like this:

````diff
diff --git a/configure.ac b/configure.ac
index c9ac073..879c1bc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -145,10 +145,6 @@ if test x"$with_gtk4" != xno; then
        PKG_CHECK_MODULES(LIBNMA_GTK4, libnma-gtk4 >= 1.8.33)
 fi
 
-PKG_CHECK_MODULES(WEBKIT, webkit2gtk-4.1, [wekbit=4.1],
-                         [PKG_CHECK_MODULES(WEBKIT, webkit2gtk-4.0,
-                         [webkit=4.0], AC_MSG_ERROR(Neither webkit2gtk-4.0 nor wekit2gtk-4.1 found))])
-
 PKG_CHECK_MODULES(LIBNM, libnm >= 1.2.0)
 LIBNM_CFLAGS="$LIBNM_CFLAGS -DNM_VERSION_MIN_REQUIRED=NM_VERSION_1_2"
 LIBNM_CFLAGS="$LIBNM_CFLAGS -DNM_VERSION_MAX_ALLOWED=NM_VERSION_1_4"
@@ -166,6 +162,9 @@ else
 fi
 if test x"$with_authdlg" != xno; then
        PKG_CHECK_MODULES(OPENCONNECT, openconnect >= 3.02)
+       PKG_CHECK_MODULES(WEBKIT, webkit2gtk-4.1, [wekbit=4.1],
+                                       [PKG_CHECK_MODULES(WEBKIT, webkit2gtk-4.0,
+                                       [webkit=4.0], AC_MSG_ERROR(Neither webkit2gtk-4.0 nor wekit2gtk-4.1 found))])
 fi
 AM_CONDITIONAL(WITH_AUTHDLG, test x"$with_authdlg" != xno)
````

Want to try to submit that upstream?

Reply:
Issue submitted to upstream: https://gitlab.gnome.org/GNOME/NetworkManager-openconnect/-/issues/99
Patch added in https://github.com/gentoo/gentoo/commit/3bfa8c0c5bf44bea1bc39e1f98ae727af023b99b
Thanks!